File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_04_databinding/ui/state/ListViewModel.java


package com.kunminx.jetpack_java.sample_04_databinding.ui.state;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.kunminx.jetpack_java.common_data.bean.Moment;
import com.kunminx.jetpack_java.sample_03_viewmodel.domain.MomentRequest;
import com.kunminx.jetpack_java.sample_03_viewmodel.domain.Request;

import java.util.List;

/**
 * Create by KunMinX at 2020/5/30
 */
public class ListViewModel extends ViewModel implements Request.IMomentRequest {

    public final MutableLiveData<List<Moment>> list = new MutableLiveData<>();

    public final MutableLiveData<Boolean> autoScrollToTopWhenInsert = new MutableLiveData<>();

    private MomentRequest mMomentRequest = new MomentRequest();

    {
        autoScrollToTopWhenInsert.setValue(true);
    }

    @Override
    public LiveData<List<Moment>> getListMutableLiveData() {
        return mMomentRequest.getListMutableLiveData();
    }

    @Override
    public void requestList() {
        mMomentRequest.requestList();
    }
}

File: jetpack_kotlin/src/main/java/com/example/jetpack_kotlin/sample_04_databinding/state/ListViewModel.kt
package com.example.jetpack_kotlin.sample_04_databinding.state

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.jetpack_kotlin.common_data.bean.Moment
import com.example.jetpack_kotlin.sample_03_viewmodel.domain.MomentRequest
import com.example.jetpack_kotlin.sample_03_viewmodel.domain.Request

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   22:03
 * description
 */
internal class ListViewModel : ViewModel(), Request.IMomentRequest {

    val list = MutableLiveData<List<Moment>>(emptyList())

    private val mMomentRequest by lazy { MomentRequest() }

    override fun getListMutableLiveData(): LiveData<List<Moment>> = mMomentRequest.getListMutableLiveData()

    override fun requestList() {
        mMomentRequest.requestList()
    }
}